formApp/src/components/form/CheckboxController.tsx

```diff
@@ -0,0 +1,53 @@
+import { Checkbox } from "@/components/ui/checkbox";
+import {
+	FormControl,
+	FormDescription,
+	FormField,
+	FormItem,
+} from "@/components/ui/form";
+import type {
+	ComponentPropsWithoutRef,
+	FC,
+	InputHTMLAttributes,
+} from "react";
+import type {
+	Control,
+	FieldValues,
+	Path,
+} from "react-hook-form";
+
+type Props<T extends FieldValues> = {
+	name: Path<T>;
+	control: Control<T>;
+} & Omit<
+	ComponentPropsWithoutRef<typeof Checkbox>,
+	"checked" | "onCheckedChange"
+> &
+	InputHTMLAttributes<HTMLInputElement>;
+export const CheckboxController: FC<
+	Props<FieldValues>
+> = ({ name, control, ...checkboxProps }) => {
+	return (
+		<FormField
+			control={control}
+			name={name}
+			render={({ field }) => (
+				<FormItem className="m-5 flex flex-row items-start space-x-3 space-y-0 rounded-md border p-4">
+					<FormControl>
+						<Checkbox
+							{...field}
+							{...checkboxProps}
+							checked={field.value}
+							onCheckedChange={(checked) => {
+								field.onChange(checked);
+							}}
+						/>
+					</FormControl>
+					<FormDescription>
+						Chama or No Chama?
+					</FormDescription>
+				</FormItem>
+			)}
```

Comment:
zbytečně uvnitř propsy render, udělej to jako children místo toho

<>{(field) => <FormItem ..... .... >}</>

Reply:
Nevím jestli tomu rozumím správně jak to myslíš, ale pokud ano, tak FormField nepodporuje children
Type '{ children: ({ field }: { field: any; }) => Element; control: Control<FieldValues>; name: string; }' is not assignable to type 'IntrinsicAttributes & { render: ({ field, fieldState, formState, }: { field: ControllerRenderProps<FieldValues, string>; fieldState: ControllerFieldState; formState: UseFormStateReturn<...>; }) => ReactElement<...>; } & UseControllerProps<...>'.
  Property 'children' does not exist on type 'IntrinsicAttributes & { render: ({ field, fieldState, formState, }: { field: ControllerRenderProps<FieldValues, string>; fieldState: ControllerFieldState; formState: UseFormStateReturn<...>; }) => ReactElement<...>; } & UseControllerProps<...>'.ts(2322)
  `return (
		<FormField control={control} name={name}>
			{({ field }) => (
				<>
					<FormItem className="m-5 flex flex-row items-start space-x-3 space-y-0 rounded-md border p-4">
						<FormControl>
							<Checkbox
								{...field}
								{...checkboxProps}
								checked={field.value}
								onCheckedChange={(checked) => {
									field.onChange(checked);
								}}
							/>
						</FormControl>
						<FormDescription>
							Chama or No Chama?
						</FormDescription>
					</FormItem>
				</>
			)}
		</FormField>
	);`